### ISSUE TITLE
Some titles on IMDb do not match the results from OMDb

### ISSUE THREAD

**Issue by blairg23** [omdbapi/OMDb-API#218]
Example, a search for the title with IMDb id `tt13061914` which is titled `Locked Down` does not match the result from OMDb which is `Lockdown`.

IMDb URL: https://www.imdb.com/title/tt13061914/

Example request: 

```
http://www.omdbapi.com?i=tt13061914&plot=full&apikey=<api_key>
```

Example response:

```
{
    "Title": "Lockdown",
    "Year": "2021",
    "Rated": "R",
    "Released": "01 Jan 2021",
    "Runtime": "N/A",
    "Genre": "Comedy, Romance",
    "Director": "Doug Liman",
    "Writer": "Steven Knight",
    "Actors": "Anne Hathaway, Chiwetel Ejiofor, Ben Stiller, Lucy Boynton",
    "Plot": "A couple attempt a high-risk, high-stakes jewellery heist at a department store.",
    "Language": "English",
    "Country": "UK",
    "Awards": "N/A",
    "Poster": "N/A",
    "Ratings": [],
    "Metascore": "N/A",
    "imdbRating": "N/A",
    "imdbVotes": "N/A",
    "imdbID": "tt13061914",
    "Type": "movie",
    "DVD": "N/A",
    "BoxOffice": "N/A",
    "Production": "N/A",
    "Website": "N/A",
    "Response": "True"
}
```

**Comment by brandan-schmitz** [comment]
I am also seeing this issue. It seems to be related to the fact that some entries have a title and then also have an "original title". This API seems to be returning the original title field when ideally it should return the main one listed.

This can be seen on the [Iron Man 3](https://www.imdb.com/title/tt1300854) movie. You can see it shows the title as `Iron Man 3` however below that is shows the original title was `Iron Man Three` which is what the API is returning:

![image](https://user-images.githubusercontent.com/6267549/127084563-4e137d7d-6be8-4c02-b8e9-c6ee72b9598f.png)

<br />
<br />

```
http://www.omdbapi.com?i=tt1300854&apikey=<api_key>
```

```json
{
  "title": "Iron Man Three",
  "year": "2013",
  "released": "03 May 2013",
  "season": null,
  "episode": null,
  "runtime": "130 min",
  "genre": "Action, Adventure, Sci-Fi",
  "director": "Shane Black",
  "writer": "Drew Pearce, Shane Black, Stan Lee",
  "actors": "Robert Downey Jr., Guy Pearce, Gwyneth Paltrow",
  "plot": "When Tony Stark's world is torn apart by a formidable terrorist called the Mandarin, he starts an odyssey of rebuilding and retribution.",
  "language": "English",
  "country": "United States",
  "awards": "Nominated for 1 Oscar. 20 wins & 63 nominations total",
  "imdbID": "tt1300854",
  "seriesID": null,
  "type": "movie",
  "response": "True",
  "error": null,
  "poster": "https://m.media-amazon.com/images/M/MV5BMjE5MzcyNjk1M15BMl5BanBnXkFtZTcwMjQ4MjcxOQ@@._V1_SX300.jpg"
}
```

**Comment by blairg23** [comment]
It's not an easy problem to solve. We need to use a fuzzy searching algorithm and perhaps some Levenshtein Distance calculation to show similarities between the title being searched and the actual title. I use something similar in my own personal movie naming app. https://en.wikipedia.org/wiki/Levenshtein_distance

**Comment by brandan-schmitz** [comment]
While I do not know the specifics of how the internals of this API work, I do not think you would need to do anything of the sort on the API end. When requesting something from the API in this case (utilizing the `i` parameter), we are requesting it by a specific IMDb ID. The data that is returned on IMDb should follow a set pattern and the API would simply be interpreting this. For this type of request the API should not need to do any searching. The issue here is specific to the fact that even when selecting an item directly though its IMDb ID it is still returning the wrong title.

I will note that you are correct in that, if one is searching for a movie then some sort of searching will need to be done. However, this recent change on IMDb to list the older title and the new ones I believe is causing the API to not return results as it should. I used to be able to do a search for `Iron Man 3` using the `s` parameter and it would return Iron Man 3 as the first movie. Now in order for it to show up you need to search for `Iron Man Three`. Again, I believe this is due to the way the API interprets data from the IMDb site and their recent change has caused issues here.

**Comment by blairg23** [comment]
@brandan-schmitz Yeah I should've been specific. If you search by title for `Locked Down`, you are not going to find `Lockdown` (the result coming back from OMDB) even though the title is `Locked Down` in IMDB.

**Comment by tgreve15** [comment]
Omdbapi draws from wikidata last I heard and not imdb directly. Wikidata is updated by the community (including potentially you). Imdb charges a lot of money to use its APIs or bulk data download compared to this free option. As such, do you have any suggestions for how to accomplish what you desire here?

**Comment by blairg23** [comment]
> Omdbapi draws from wikidata last I heard and not imdb directly. Wikidata is updated by the community (including potentially you). Imdb charges a lot of money to use its APIs or bulk data download compared to this free option. As such, do you have any suggestions for how to accomplish what you desire here?

You mean these incredibly expensive bulk downloads which are updated daily? https://datasets.imdbws.com/

**Comment by tgreve15** [comment]
Excellent, nice find, glad to see that option is still available from IMDB. I forgot they distinguished between the expensive option for commercial use and free for non-commercial, so when I saw the cost for the downloads I gathered that they no longer had the free option. I'm also glad to see you have a backup option in case what is in Wikidata, that I believe Omdbapi pulls from, isn't accurate enough. Thanks for finding and sharing that link.

**Comment by blairg23** [comment]
No problem! Glad to help!
